spec/policy/headers/headers_spec.lua

```diff
@@ -70,6 +96,31 @@ describe('Headers policy', function()
 
         assert.same('2', ngx.req.get_headers()[header])
       end)
+
+      describe('when no value is provided', function()
+        describe('and the header is not set', function()
+          it('does not create it', function()
+            local config = { [request_headers] = { { op = set, header = header } } }
+            local headers_policy = HeadersPolicy.new(config)
+
+            headers_policy:rewrite()
+
+            assert.is_nil(ngx.req.get_headers()[header])
+          end)
+        end)
+
+        describe('and the header is already set', function()
+          it('deletes it', function()
```

Comment:
I wasn't totally sure about this one but decided to keep it working as it is now.